homeassistant/components/homematicip_cloud/alarm_control_panel.py

```diff
@@ -95,10 +96,19 @@ def supported_features(self) -> int:
         """Register callbacks."""
         self._home.on_update(self._async_device_changed)
 
+    @callback
```

Comment:
Why does it matter if we decorate this method? Isn't this callback called from the device library in the `home` instance?

Reply:
it makes no difference, but i thought it has to be there. should it be removed?

Reply:
It doesn't hurt, but it has no effect.

Reply:
It's good to have decorators on all methods that we expect to be only called inside the event loop. It helps understand the code.

Reply:
We will never be able to do that everywhere. I don't think we should try.

Reply:
We actually already do everywhere. See:

```
git grep -B1 "  def async_" | grep -v 'callback' | grep -v '\-\-' | grep -v ' def '
git grep -B1 "  def _async_" | grep -v 'callback' | grep -v '\-\-' | grep -v ' def '
```

I think that it makes it very clear for people reading the code about what kind of context the method can/should be called.

Reply:
We'll never decorate other library functions 100%. We won't be able to trust that this decorator is the truth about this. You have to read other things in the code to find out.

It's a war we cannot win. I think we should pick our battles.